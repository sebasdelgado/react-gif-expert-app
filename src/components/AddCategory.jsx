import { useState } from "react";

// export const AddCategory = ( {setCategories} ) => { //De esta manera se recibe la función
export const AddCategory = ( {onNewCategory} ) => {

    const [ inputValue, setInputValue ] = useState('');

    //Desestructuramos el event y obetenemos el target
    const onInputChange = ({ target }) => {
        setInputValue( target.value );
    }

    
    const onSubmit = ( event ) => {
        //Usamos una función para cuando se ejecute el submit 
        //no refresque el navegador, debido a que el submit del formulario
        //refresca el navegador
        event.preventDefault();

        if(inputValue.trim().length <= 1) return;

        // setCategories( categories => [ inputValue, ...categories]); //De esta manera agregamos el valor
                                                                        //cuando recibimos la función
        onNewCategory( inputValue.trim() );
        //Limpiamos el input
        setInputValue('');
    }

    return (

        <form onSubmit={ (event) => onSubmit(event) }>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={ inputValue }
                // Hacemos uso del onChange porque si seteamos un value
                // este no nos va a dejar modificar cuando se renderice 
                onChange = { onInputChange}
            />
        </form>
    )
}