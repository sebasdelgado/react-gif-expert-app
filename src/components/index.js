//Archivo barril para manejar las expotaciones
export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';