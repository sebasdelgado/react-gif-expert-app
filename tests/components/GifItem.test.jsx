import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem/>', () => {

    const title = 'Prueba';
    const url = 'https://prueba.com/prueba.jpg';

    test('Debe hacer match con el snapshot', () => {
       
        const { container }= render( <GifItem title = { title } url = { url } />);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con la URL y el ALT indicado', () => {
       
        render( <GifItem title = { title } url = { url } />);
        //screen.debug() nos permite ver el componente el consola
        // screen.debug();
        //Desestructuramos los atributos del img
        //getByRole() nos permite obtener elementos de html por su tag
        const { src, alt } = screen.getByRole('img');
        //toBe( ) comparamos los valores
        expect ( src ).toBe( url );
        expect ( alt ).toBe( title );
    });

    test('Debe mostrar el tÃ­tulo del componente', () => {
       
        render( <GifItem title = { title } url = { url } />);
        // toBeTruthy() nos dice si el texto exite
        expect ( screen.getByText( title ) ).toBeTruthy();
    });
});