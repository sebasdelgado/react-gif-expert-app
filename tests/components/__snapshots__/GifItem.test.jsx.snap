// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Prueba"
      src="https://prueba.com/prueba.jpg"
    />
    <p>
      Prueba
    </p>
  </div>
</div>
`;
